File: carp.dsp.core/src/jvmTest/resources/scripts/step1_load.R
#!/usr/bin/env Rscript
# Step 1: Load data.
# Reads input CSV and adds a row_id column.
# Usage: step1_load.R <input_file> <output_file>

args <- commandArgs(trailingOnly = TRUE)

if (length(args) < 2) {
  stop("Usage: step1_load.R <input_file> <output_file>")
}

input_file  <- args[1]
output_file <- args[2]

dir.create(dirname(output_file), showWarnings = FALSE, recursive = TRUE)

data <- read.csv(input_file, stringsAsFactors = FALSE)
data <- cbind(row_id = seq_len(nrow(data)), data)

write.csv(data, output_file, row.names = FALSE)
message(sprintf("Loaded %d rows -> %s", nrow(data), output_file))

File: carp.dsp.core/src/jvmTest/resources/scripts/step1_load.py
#!/usr/bin/env python
"""
Step 1: Load data.
Reads input CSV and adds a row_id column.
Usage: step1_load.py <input_file> <output_file>
"""

import sys
import csv
from pathlib import Path


def load(input_file: str, output_file: str) -> None:
    input_path = Path(input_file)
    output_path = Path(output_file)
    output_path.parent.mkdir(parents=True, exist_ok=True)

    with open(input_path, 'r', newline='') as infile:
        reader = csv.DictReader(infile)
        rows = list(reader)
        fieldnames = list(reader.fieldnames or [])

    fieldnames = ['row_id'] + fieldnames
    for i, row in enumerate(rows):
        row['row_id'] = str(i + 1)

    with open(output_path, 'w', newline='') as outfile:
        writer = csv.DictWriter(outfile, fieldnames=fieldnames)
        writer.writeheader()
        writer.writerows(rows)

    print(f"Loaded {len(rows)} rows -> {output_file}")


if __name__ == '__main__':
    if len(sys.argv) != 3:
        print("Usage: step1_load.py <input_file> <output_file>")
        sys.exit(1)
    load(sys.argv[1], sys.argv[2])
